ui/src/main/js/page/stocks/index.js

```diff
@@ -0,0 +1,180 @@
+    }
+  }
+
+  searchByCompanyName(companyName) {
+      const navUri = root+'/stocks/search/findByCompanyNameStartsWith?companyName='+companyName.toUpperCase();
+      this.searchForStockCollection(navUri);
+  }
+
+  searchForStockCollection(navUri) {
+      client({
+          method: 'GET',
+          path: navUri
+      }).then(stockCollection => {
+          this.links = stockCollection.entity._links;
+          this.page = stockCollection.entity.page;
+          return stockCollection.entity._embedded.stocks.map(stock =>
+                  client({
+                      method: 'GET',
+                      path: stock._links.self.href
+                  })
+          );
+      }).then(stockPromises => {
+          return when.all(stockPromises);
+      }).then(stocks => {
+          this.setState({
+              page: this.page,
+              stocks: stocks,
+              attributes: Object.keys(this.schema.properties),
+              pageSize: this.state.pageSize,
+              links: this.links
+          });
+      });
+  }
+
+  searchBySymbol(symbol) {
+      const navUri = root+'/stocks/search/findBySymbol?symbol='+symbol.toUpperCase();
+      client({
+          method: 'GET',
+          path: navUri
+      }).then(stock => {
+          this.setState({
+              page: this.page,
+              stocks: [stock],
+              pageSize: this.state.pageSize,
+              links: this.links
+          });
+      });
+  }
+
+  refreshCurrentPage(message) {
+    let updatedStock = message.body;//JSON.parse(message.body);
+  	follow(client, root, [{
+  		rel: 'stocks',
+  		params: {size: this.state.pageSize, sort:'percentageChange,desc', page: this.state.page.number}
+  	}]).then(stockCollection => {
+              this.links = stockCollection.entity._links;
+              this.page = stockCollection.entity.page;
+              return stockCollection.entity._embedded.stocks.map(stock =>
+                      client({
+                          method: 'GET',
+                          path: stock._links.self.href
+                      })
+              );
+          }).then(stockPromises => {
+              return when.all(stockPromises);
+          }).then(stocks => {
+              let stock = this.state.stocks.find(
+                  x => x.url.includes(updatedStock));
+                  if(typeof stock !== 'undefined') {
+                    stock._changeEvent = 'increase';
+      //              if (parseInt(updatedStock.price) > parseInt(stock.price)) {
+      //                updatedStock._changeEvent = 'increase';
+      //
+      //              } else if (parseInt(updatedStock.price) < parseInt(stock.price)) {
+      //                updatedStock._changeEvent = 'decrease';
+      //              }
+                    this.setState({
+                        page: this.page,
+                        stocks: this.state.stocks.map(
+                                          x => x.url.includes(updatedStock)
```

Comment:
Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/modernmaster/forecaster/pullRequest?prid=7238741)
```suggestion
                                          (x) => x.url.includes(updatedStock)
```